chia/data_layer/data_layer_server.py

```diff
@@ -0,0 +1,64 @@
+import aiosqlite
+from aiohttp import web
+from chia.data_layer.data_store import DataStore
+from chia.util.db_wrapper import DBWrapper
+from chia.types.blockchain_format.tree_hash import bytes32
+from tests.core.data_layer.util import generate_big_datastore
+
+
+class DataLayerServer:
+    async def handle_tree_root(self, request: web.Request) -> web.Response:
+        tree_id = request.rel_url.query["tree_id"]
+        tree_id_bytes = bytes32.from_hexstr(tree_id)
+        tree_root = await self.data_store.get_tree_root(tree_id_bytes)
+        result = {
+            "tree_id": tree_id,
+            "generation": tree_root.generation,
```

Comment:
is "generation" a counter we increment by 1 every time we update the singleton?

Reply:
it increments by 1 every time we insert or delete a single node - as far as I understand the code.

Reply:
I guess a caveat to this is that if the data propagation allows for getting arbitrary generations then we'll have a sparsely populated list of generations on some clients.